site/content/en/docs/Installation/upgrading.md

```diff
@@ -51,6 +51,74 @@ The following are steps to implement this:
 for the period of your upgrade, as there will be a short period in which Agones will be non-responsive during the upgrade.
 
+{{% feature publishVersion="1.43.0" %}}
+#### In-Place Agones Upgrades
+
+{{< alert color="warning" title="Warning" >}}
+Work is ongoing for [In-Place Agones Upgrades](https://github.com/googleforgames/agones/issues/3766),
+and the feature is currently in `Dev`. Please continue to use the multi-cluster strategy for
+production critical upgrades. Feedback on the upcoming `Alpha` and `Beta` releases is appreciated.
+{{< /alert >}}
+
+For In-Place Agones Upgrades we highly recommend installing using Helm. Helm has a significant
+advantage over `install.yaml` in that Helm automatically rolls back the upgrade if the agones-system
+pods are not all successfully running within the designated `--timeout`.
+
+Note: By “controller derived configuration” we mean the parts of the Game Server’s final state that
+are not part of the Game Server spec template that are instead passed to the Game Server through the
+controller, such as the FEATURE_GATES or the Agones SDK Image.
+
+##### Upgrade Existing Fleet (recommended)
+
+1. Run `helm upgrade my-release --install --atomic --wait --timeout 10m --namespace=agones-system`
+with all the appropriate arguments, such a `--version`, for your specific upgrade. Keep in mind that
+`helm upgrade` overwrites all `--set agones.` arguments, so these must be set for each upgrade. See
+[Helm Configuration]({{< relref "./Install Agones/helm.md" >}}) for a list of all available
+configurable parameters.
+2. Wait until the `helm upgrade` is complete.
+3. Kick off rolling update of existing Fleet.
+    1. Make any change to the Fleet Spec Template in your fleet.yaml and reapply with
+    `kubectl apply -f fleet.yaml`.
+    2. If you have no changes to make to the existing Fleet Spec Template in the fleet.yaml,
+    then add a label to the spec.template.metadata.labels. This has the added advantage that Game
+    Servers can be selected for allocation by label. Reapply with `kubectl apply -f fleet.yaml`.
+    ```yaml
+    apiVersion: agones.dev/v1
+    kind: Fleet
+    metadata:
+      name: example-fleet
+    spec:
+      replicas: 2
+      template:
+        metadata:
+          # Adding a label will start a Fleet rollout (RollingUpdate or Recreate based on the replacement strategy type) with the most up to date Agones config. The label can be any `key: value` pair.
+          labels:
+            release: 1.42.0
+    ```
+
+    3. This creates a new Game Server Set. Ready Game Servers will shut down and be recreated on
+    the new configuraiton on the new Game Server Set. Allocated Game Servers will use the existing
+    configuration, and once they are set back to Ready, these Game Servers will shut down and be
+    replaced by Ready Game Servers at the new configuration on the new Game Server Set.
+    4. The rollout is complete once there is one Game Server Set at the new configuration.
```

Comment:
```suggestion
4.Wait until all the Ready `GameServers` are replaced.
```

Original statement is untrue, and also the end user rarely looks at GameServerSets, 

If there are allocated `GameServer` you could get multiple `GameServerSets`.

Reply:
Allocated game servers will still be at the old configuraiton until they are set back to ready, shut down, etc. So all the Ready game servers could be at the new configuration, and there would still be game servers at the old configuraiton. I guess, what are we defining as "the rollout is complete"? It could be either that Agones is upgraded (this should be in the ~minutes), or all Ready game servers are at the new configuraiton, or all game servers are at the new configuraiton?

Reply:
In regards to when a _Fleet_ rollout is complete:

```
"Done" (for the initial Alpha release) is:

At least one Ready or Allocated or Reserved GameServer from the new GameServerSet.
Only Allocated and Reserved GameServers on any previous GameServerSets.
```
https://github.com/googleforgames/agones/issues/2817#issuecomment-1526003786

Should this apply to the Agones upgrade as well?

Reply:
I'd be tempted to be a little more ambiguous here. Probably the best we could do would be "Wait until all the previous Ready GameServers have been replaced by the new version.'

And leave it up to the end user what they want to do with anything Allocated or Reserved.